File: flutter-app/lib/features/vendors/domain/vendor.dart
/// A marketplace vendor (mirrors vendor-service `VendorDto`). EOS-002-P3-Part-06.
class Vendor {
  const Vendor({
    required this.id,
    required this.businessName,
    required this.category,
    required this.city,
    required this.latitude,
    required this.longitude,
    required this.serviceRadiusKm,
    required this.rating,
    required this.verificationLevel,
    required this.portfolio,
    this.description,
    this.address,
    this.yearsOfExperience,
    this.teamSize,
    this.startingPrice,
    this.reviewCount,
    this.trustScore,
    this.responseTimeHours,
    this.workingDays,
  });

  final String id;
  final String businessName;
  final String category;
  final String? description;
  final String city;
  final String? address;
  final double latitude;
  final double longitude;
  final double serviceRadiusKm;
  final int? yearsOfExperience;
  final int? teamSize;
  final num? startingPrice;
  final double rating;
  final int? reviewCount;

  /// 0–4 (EOS-002-P3-Part-06 §6). 4 = HiLo Verified Partner.
  final int verificationLevel;
  final double? trustScore;
  final double? responseTimeHours;
  final List<String> portfolio;
  final List<String>? workingDays;

  String get verificationLabel => switch (verificationLevel) {
    4 => 'HiLo Verified Partner',
    3 => 'Documents verified',
    2 => 'Mobile verified',
    1 => 'Email verified',
    _ => 'Unverified',
  };
}

/// Curated launch categories (kept in sync with vendor-service).
const List<String> kVendorCategories = <String>[
  'Catering',
  'Decoration',
  'Photography',
  'Videography',
  'Entertainment',
  'Music',
  'Beauty',
  'Event Services',
  'Transportation',
  'Gifts',
];

File: backend-api/services/vendor-service/src/domain/vendor.ts
import type { DocumentMetadata } from '@hilo/backend-shared';

/**
 * Vendor verification levels (EOS-002-P3-Part-06 §6). Level 4 = HiLo Verified Partner.
 * Levels are granted by the backend/admin only — never self-assigned.
 */
export const VerificationLevel = {
  None: 0,
  EmailVerified: 1,
  MobileVerified: 2,
  DocumentsVerified: 3,
  HiLoVerifiedPartner: 4,
} as const;

export type VerificationLevel = (typeof VerificationLevel)[keyof typeof VerificationLevel];

export function isVerificationLevel(value: unknown): value is VerificationLevel {
  return typeof value === 'number' && Number.isInteger(value) && value >= 0 && value <= 4;
}

/** Curated launch categories (EOS-002-P3-Part-06 §3). */
export const VENDOR_CATEGORIES = [
  'Catering',
  'Decoration',
  'Photography',
  'Videography',
  'Entertainment',
  'Music',
  'Beauty',
  'Event Services',
  'Transportation',
  'Gifts',
] as const;

/**
 * The Vendor aggregate root (owner: vendor-service). Carries the mandatory metadata
 * envelope (EOS-000 §35). `ownerUid` is the business account that manages this listing.
 */
export interface Vendor extends DocumentMetadata {
  ownerUid: string;
  businessName: string;
  category: string;
  description?: string;
  contactPerson?: string;
  city: string;
  address?: string;
  latitude: number;
  longitude: number;
  /** How far the vendor travels to serve, km (§5 Location). */
  serviceRadiusKm: number;
  yearsOfExperience?: number;
  teamSize?: number;
  /** Indicative starting price, INR. */
  startingPrice?: number;
  /** Google rating 0–5 (read-only). */
  rating: number;
  reviewCount?: number;
  verificationLevel: VerificationLevel;
  /** HiLo Trust Score 0–1 (calculated internally — see trust-score.ts). */
  trustScore?: number;
  reviewSentiment?: number;
  /** Average first-response time in hours (drives scoring). */
  responseTimeHours?: number;
  /** 0–1 fraction of confirmed bookings later cancelled by the vendor. */
  cancellationRate?: number;
  /** 0–1 fraction of customers who book again. */
  repeatBookingRate?: number;
  /** 0–1 — how ready the vendor is for new work right now. */
  availabilityScore?: number;
  portfolio: string[];
  workingDays?: string[];
}

/** Input a business user provides at onboarding (EOS-002-P3-Part-06 §5). */
export interface VendorOnboardingInput {
  businessName: string;
  category: string;
  description?: string;
  contactPerson?: string;
  city: string;
  address?: string;
  latitude: number;
  longitude: number;
  serviceRadiusKm?: number;
  yearsOfExperience?: number;
  teamSize?: number;
  startingPrice?: number;
  workingDays?: string[];
}
